dashboard/lib/pd/survey_pipeline/daily_survey_decorator.rb

```diff
@@ -37,45 +37,83 @@ def self.decorate(summary_data:, parsed_data:)
         facilitator_response_counts: {}
       }
 
-      # Build non-hidden question list result[:questions]
-      # Hash{form_name => {general => {question_name => question_content}}}
-      questions = parsed_data[:questions]
-      questions.each do |form_id, form_questions|
-        form_name = get_form_name(form_id)
+      question_by_names = index_question_by_names(parsed_data[:questions])
 
-        result[:questions][form_name] ||= {general: {}}
-        dest_collection = result[:questions][form_name][:general]
-
-        form_questions.each do |_, qcontent|
-          next if qcontent[:hidden]
-          dest_collection[qcontent[:name]] = qcontent.except(:name)
-        end
-      end
-
-      # Populate results for result[:this_workshop]
-      # Hash{form_name => {response_count => number, general => {question_name => summary_result}}}
+      # Populate entries for result[:this_workshop] and result[:questions]
       summary_data.each do |summary|
-        # Only process summarization for specific workshops and forms
+        # Only process summarization for specific workshops and forms.
         form_id = summary[:form_id]
         next unless summary[:workshop_id] && form_id
 
+        form_name = get_form_short_name(form_id)
         workshop_id = summary[:workshop_id]
-        form_name = get_form_name(form_id)
+        q_name = summary[:name]
 
+        # Create top structures if not already created
         result[:course_name] ||= Pd::Workshop.find_by_id(workshop_id)&.course
         result[:this_workshop][form_name] ||= {
           response_count: parsed_data[:submissions][form_id].size,
-          general: {}
+          general: {},
+          facilitator: {}
         }
+        result[:questions][form_name] ||= {general: {}, facilitator: {}}
 
-        result[:this_workshop][form_name][:general][summary[:name]] = summary[:reducer_result]
+        # Create an entry in result[:this_workshop].
+        # General format:
+        # Hash{form_name => {response_count => number, general => {question_name => summary_result}}}
+        #
+        # Format for facilicator-specific result:
+        # Hash{form_name => {response_count => number,
+        #   facilitator => {question_name => {factilitator_name => summary_result}}}}
+        if summary[:facilitator_id]
```

Comment:
does this take into account permissions, that co-facilitators should not see each other's results?

Reply:
+1

Reply:
oh great point, it does not. Will add

Reply:
Added permission filter and some unit tests. I may need help to test an end-to-end scenario locally.